### ISSUE TITLE
Unable to reproduce DanceTrack val results — trained model performs far worse than pretrained

### ISSUE THREAD

**Issue by ShenZheng2000** [ymzis69/HybridSORT#52]
I tested your pretrained HybridSORT model on DanceTrack using:
```
exp_name="hybridsort_dancetrack_val"
python tools/run_hybrid_sort_dance.py -f exps/example/mot/yolox_dancetrack_val_hybrid_sort.py -b 1 -d 1 --fp16 --fuse --expn $exp_name 
```

This gives (HOTA = 59.41, IDF1 = 60.98):
```
HOTA DetA AssA DetRe DetPr AssRe AssPr LocA RHOTA HOTA(0) LocA(0) HOTALocA(0) MOTA MOTP MODA CLR_Re CLR_Pr MTR PTR MLR CLR_TP CLR_FN CLR_FP IDSW MT PT ML Frag sMOTA IDF1 IDR IDP IDTP IDFN IDFP Dets GT_Dets IDs GT_IDs
59.405 78.92 44.918 85.034 88.886 50.914 69.171 91.093 61.784 66.892 88.251 59.033 89.502 90.236 90.257 92.961 97.173 86.813 12.088 1.0989 209300 15848 6089 1699 237 33 3 3266 80.425 60.98 59.658 62.361 134319 90829 81070 215389 225148 692 273
```

I then trained the model myself using the provided training script:
```
python3 tools/train.py -f exps/example/dancetrack/yolox_x.py -d 8 -b 48 --fp16 -o -c pretrained/yolox_x.pth
```

and evaluated it with the same evaluation command, which gives (HOTA = 53.32, IDF1 = 53.98)
```
HOTA DetA AssA DetRe DetPr AssRe AssPr LocA RHOTA HOTA(0) LocA(0) HOTALocA(0) MOTA MOTP MODA CLR_Re CLR_Pr MTR PTR MLR CLR_TP CLR_FN CLR_FP IDSW MT PT ML Frag sMOTA IDF1 IDR IDP IDTP IDFN IDFP Dets GT_Dets IDs GT_IDs
53.318 76.922 37.114 84.018 86.526 42.918 65.508 89.933 55.83 61.323 86.784 53.219 88.361 88.957 89.2 93.151 95.931 89.377 9.5238 1.0989 209727 15421 8895 1888 244 26 3 3370 78.075 53.982 53.199 54.787 119777 105371 98845 218622 225148 842 273
```

This gap (~6 HOTA, ~7 IDF1) is far larger than what can be explained by seed variation.

Is there any missing training detail, config mismatch, or evaluation assumption needed to reproduce the reported DanceTrack results?

**Comment by ShenZheng2000** [comment]
@ymzis69 

I just regenerated coco annotations and rerun the experiments. 

Here is the pretrained model’s detector performance (from `val_log.txt`). I also compared it with OC-SORT’s retrained / pretrained detector and did not observe significant differences.
```
Average forward time: 43.51 ms, Average track time: 5.59 ms, Average inference time: 49.11 ms
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.795
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.947
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.846
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.667
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.801
```

This is very similar to the re-trained model’s detector performance below (from `val_log.txt`):
```
Average forward time: 31.54 ms, Average track time: 3.52 ms, Average inference time: 35.06 ms
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.797
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.957
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.847
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.666
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.804
```

**Overall, the detector quality is almost identical between the pretrained and re-trained models. However, the tracking results (from `pedestrian_summary.txt`) show significant differences.**

Below are the tracking results using the **pretrained** detector:
```
HOTA DetA AssA DetRe DetPr AssRe AssPr LocA RHOTA HOTA(0) LocA(0) HOTALocA(0) MOTA MOTP MODA CLR_Re CLR_Pr MTR PTR MLR CLR_TP CLR_FN CLR_FP IDSW MT PT ML Frag sMOTA IDF1 IDR IDP IDTP IDFN IDFP Dets GT_Dets IDs GT_IDs
58.995 78.985 44.27 85.074 88.928 50.073 68.861 91.104 61.349 66.334 88.287 58.564 89.501 90.235 90.262 92.964 97.175 86.447 12.454 1.0989 209306 15842 6084 1713 236 34 3 3268 80.422 60.146 58.842 61.508 132482 92666 82908 215390 225148 697 273
```

Below are the tracking results using the **retrained** detector:
```
HOTA DetA AssA DetRe DetPr AssRe AssPr LocA RHOTA HOTA(0) LocA(0) HOTALocA(0) MOTA MOTP MODA CLR_Re CLR_Pr MTR PTR MLR CLR_TP CLR_FN CLR_FP IDSW MT PT ML Frag sMOTA IDF1 IDR IDP IDTP IDFN IDFP Dets GT_Dets IDs GT_IDs
57.373 79.165 41.772 85.199 89.008 47.613 67.793 91.092 59.638 64.523 88.284 56.963 89.968 90.253 90.713 93.217 97.384 87.179 11.722 1.0989 209876 15272 5637 1677 238 32 3 3160 80.883 57.671 56.437 58.96 127067 98081 88446 215513 225148 714 273
```

For reference, I list the command lines I used below. 
```
# test-with-pretrained
exp_name="hybridsort_dancetrack_val"
python tools/run_hybrid_sort_dance.py \
    -f exps/example/mot/yolox_dancetrack_val_hybrid_sort.py \
    -b 1 -d 1 --fp16 --fuse --expn ${exp_name}

# retrain
exp_name="yolox_dancetrack_val_hybrid_sort_2_8"
python tools/train.py \
  -f exps/example/mot/yolox_dancetrack_val_hybrid_sort.py \
  -d 8 -b 48 --fp16 -o -c pretrained/yolox_x.pth \
  -expn $exp_name

# test-with-retrained
exp_name="yolox_dancetrack_val_hybrid_sort_2_8"
python tools/run_hybrid_sort_dance.py \
    -f exps/example/mot/yolox_dancetrack_val_hybrid_sort.py \
    -b 1 -d 1 --fp16 --fuse --expn ${exp_name} \
    -c YOLOX_outputs/${exp_name}/best_ckpt.pth.tar
```
